lib/repeatable/expression/base.rb

```diff
@@ -24,6 +24,29 @@ def to_h
         )
       end
 
+      def union(other)
+        if other.is_a?(Union)
+          other.union(self)
+        else
+          Union.new(self, other)
+        end
+      end
+      alias + union
```

Comment:
While you're here, would you mind also adding `|` as an alias like we have in Ruby's [`Set`](http://ruby-doc.org/stdlib-2.3.1/libdoc/set/rdoc/Set.html#method-i-7C)?